settings/tui/screen/elements/buttons.go

```diff
@@ -0,0 +1,115 @@
+package elements
+
+import (
+	"github.com/gdamore/tcell"
+	"errors"
+	"github.com/rivo/tview"
+)
+
+const totalCols = 8
+
+type ButtonsBox struct {
+	Grid *tview.Grid
+	buttons []*tview.Button
+	totalButtons int
+	currentFocus int
+}
+
+func NewButtonsBox() *ButtonsBox {
+	b := ButtonsBox {
+		tview.NewGrid(),
+		make([]*tview.Button, 0, totalCols),
+		0,
+		0,
+	}
+
+	l := make([]int, totalCols)
+	for i := range l {
+		l[i] = -1
+	}
+
+	b.Grid = b.Grid.
+		SetColumns(l...).
+		SetRows(4)
+
+	return &b
+}
+
+func (b *ButtonsBox) AddButton(text string) error {
+	if b.totalButtons == totalCols {
+		return errors.New("Maximum domains of interest reached")
+	}
+
+	btn := tview.NewButton(text)
+	btn.SetBackgroundColor(tcell.ColorGreen)
+	btn.SetBackgroundColorActivated(tcell.ColorOrangeRed)
+	b.buttons = append(b.buttons, btn)
+	idx := b.totalButtons
+	b.Grid.AddItem(btn, 0, idx, 1, 1, 0, 0, true)
+	b.totalButtons ++
+
+	return nil
+}
+
+func (b *ButtonsBox) RemoveButton() error {
+	if b.totalButtons == 0 {
+		return errors.New("No domains left to remove")
+	}
+
+	idx := b.currentFocus
+	btn := b.buttons[idx]
+
+	b.buttons = append(b.buttons[:idx], b.buttons[idx+1:]...)
```

Comment:
Too many slicing and array deconstruction in one line. Mention the info in a comment